crates/node/src/cluster_marker.rs

```diff
@@ -143,37 +160,58 @@ fn validate_and_update_cluster_marker_inner(
         });
     }
 
-    // find the next compatibility boundary wrt to the max version
-    let (_, compatibility_boundary) = compatibility_map
-        .range(..=cluster_marker.max_version.clone())
-        .next_back()
-        .expect("at least version 0.9.0 should be present");
+    // versions 0.9 and 1.0.0 don't write this field --> default to current version
+    let min_forward_compatible_version = cluster_marker
+        .min_forward_compatible_version
+        .clone()
+        .unwrap_or(cluster_marker.current_version.clone());
+
+    // check forward compatibility by comparing this version with minimal compatible version
+    if this_version.cmp_precedence(&min_forward_compatible_version) == Ordering::Less {
```

Comment:
Just a style comment: You can get the same semantics of this `cmp_precedence` using directly the usual comparison operators if you remove the build metadata field when you instantiate the `current_version` (we don't set ever set this anyway, but for safety)

Reply:
I would keep it because the method makes it clear how we compare versions.